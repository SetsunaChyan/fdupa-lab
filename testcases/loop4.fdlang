function main(){
fib0 = input();
fib1 = input();

while (fib0 >= 3) {
    fib0 = fib0 - 3;
}
while (fib1 >= 3) {
    fib1 = fib1 - 3;
}

check_interval(fib0, 0, 1);
check_interval(fib1, 0, 1);
check_interval(fib0, 0, 2);
check_interval(fib1, 0, 2);

while (i < 5) {
    tmp = fib0 + fib1;
    fib0 = fib1;
    fib1 = tmp;
    i = i + 1;
}

check_interval(fib1, 0, 26);
}
