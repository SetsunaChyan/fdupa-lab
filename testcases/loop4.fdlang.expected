Line 12:  NO; fib0 in [0, 2]
Line 13:  NO; fib1 in [0, 2]
Line 14: YES; fib0 in [0, 2]
Line 15: YES; fib1 in [0, 2]
Line 24: YES; fib1 in [0, 0] U [5, 5] U [8, 8] U [10, 10] U [13, 13] U [16, 16] U [18, 18] U [21, 21] U [26, 26]
